src/invoices/InvoiceForm/InvoiceForm.js

```diff
@@ -421,16 +490,25 @@ const InvoiceForm = ({
                         />
                       </Col>
                       <Col data-test-col-currency xs={3}>
-                        <FieldSelectionFinal
-                          dataOptions={currenciesOptions}
-                          disabled={isEditPostApproval}
-                          id="invoice-currency"
-                          labelId="ui-invoice.invoice.currency"
-                          name="currency"
-                          onChange={onChangeCurrency}
-                          required
-                          validate={validateRequired}
-                        />
+                        {isEditPostApproval
+                          ? (
+                            <KeyValue
+                              label={<FormattedMessage id="ui-invoice.invoice.currency" />}
+                              value={currency}
```

Comment:
1 It looks like a copy-paste of displaying currency;
2 I'm not sure that it's right to display currency in that way. Maybe something like that https://github.com/folio-org/stripes-components/blob/master/hooks/useCurrencyOptions/useCurrencyOptions.js#L12